server/main_server/src/rodsServer.cpp

```diff
@@ -568,13 +568,23 @@ namespace
 
             log::trace("Encrypted request.");
 
-            const auto zmq_server_target = fmt::format("tcp://localhost:{}", env.irodsCtrlPlanePort);
+            const auto zmq_server_target = fmt::format("tcp://{}:{}", env.rodsHost, env.irodsCtrlPlanePort);
             log::trace("Connecting to local control plane [{}] ...", zmq_server_target);
 
+            if (!is_control_plane_accepting_requests.load()) {
+                log::info("Cannot determine if delay server on remote host [{}] is running at this time."
+                          "Control plane is not accepting requests.", _hostname);
+
+                // The following only applies to delay server migration.
+                // Return true so that the successor does not attempt to launch the delay server.
+                // Returning std::nullopt or false can possibly lead to the delay server being launched.
+                return true;
+            }
+
             zmq::context_t zmq_ctx;
             zmq::socket_t zmq_socket{zmq_ctx, zmq::socket_type::req};
-            zmq_socket.set(zmq::sockopt::sndtimeo, 500);
-            zmq_socket.set(zmq::sockopt::rcvtimeo, 5000);
+            zmq_socket.set(zmq::sockopt::sndtimeo, 500);    // TODO Should this be a config option?
+            zmq_socket.set(zmq::sockopt::rcvtimeo, 5000);   // TODO Should this be a config option?
```

Comment:
See code comments.

These values represent the timeout values for sending/receiving messages via 0MQ.

They are in milliseconds.

Reply:
yes, please make an issue to pull those two out... so 4.3.1 or later.

are there ramifications to other machines with these two?   or can different machines have different settings and play well with each other?

Reply:
That should be fine as long as messages are sent/received within the timeout.

Reply:
Created issue: https://github.com/irods/irods/issues/6370